c++/test/TestByteRLEEncoder.cc

```diff
@@ -0,0 +1,231 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ByteRLE.hh"
+#include "MemoryOutputStream.hh"
+
+#include "wrap/gtest-wrapper.h"
+#include "wrap/orc-proto-wrapper.hh"
+
+#include <cstdlib>
+
+namespace orc {
+
+  const int DEFAULT_MEM_STREAM_SIZE = 1024 * 1024; // 1M
+
+  void generateNotNull(uint64_t numValues,
+                       uint64_t numNulls,
+                       char ** notNull) {
+    if (numNulls != 0 && notNull != nullptr) {
+      *notNull = new char[numValues];
+      memset(*notNull, 1, numValues);
+      while (numNulls > 0) {
+        uint64_t pos = static_cast<uint64_t>(std::rand()) % numValues;
+        if ((*notNull)[pos]) {
+          (*notNull)[pos] = static_cast<char>(0);
+          --numNulls;
+        }
+      }
+    }
+  }
+
+  char * generateData(uint64_t numValues,
+                      uint64_t numNulls = 0,
+                      char ** notNull = nullptr) {
+    generateNotNull(numValues, numNulls, notNull);
+    char * res = new char[numValues];
+    for (uint64_t i = 0; i < numValues; ++i) {
+        res[i] = static_cast<char>(std::rand() % 256);
+    }
+    return res;
+  }
+
+  char * generateBoolData(uint64_t numValues,
+                          uint64_t numNulls = 0,
+                          char ** notNull = nullptr) {
+    generateNotNull(numValues, numNulls, notNull);
+    char * res = new char[numValues];
+    for (uint64_t i = 0; i < numValues; ++i) {
+        res[i] = static_cast<char>(std::rand() % 2);
+      }
+    return res;
+  }
+
+  void decodeAndVerify(
+                       const MemoryOutputStream& memStream,
+                       char * data,
+                       uint64_t numValues,
+                       char* notNull) {
+
+    std::unique_ptr<SeekableInputStream> inStream(
+      new SeekableArrayInputStream(memStream.getData(), memStream.getLength()));
+
+    std::unique_ptr<ByteRleDecoder> decoder =
+      createByteRleDecoder(std::move(inStream));
+
+    char* decodedData = new char[numValues];
```

Comment:
I don't see a huge difference here - we are manipulating raw buffer anyway. I would just leave them there unless you feel strongly about it.